etl/steps/data/garden/lgbt_rights/2024-06-11/criminalization_mignot.py

```diff
@@ -0,0 +1,411 @@
+        tb=tb, start_col="former_criminalizations1", end_col="former_decriminalizations1", status="Illegal"
+    )
+    tb_previous_decriminalization_2 = expand_table(
+        tb=tb, start_col="former_decriminalizations2", end_col="former_criminalizations1", status="Legal"
+    )
+    tb_previous_criminalization_2 = expand_table(
+        tb=tb, start_col="former_criminalizations2", end_col="former_decriminalizations2", status="Illegal"
+    )
+
+    # Concatenate all tables
+    tb = pr.concat(
+        [
+            tb_last_decriminalization,
+            tb_before_last_decriminalization,
+            tb_previous_decriminalization_1,
+            tb_previous_criminalization_1,
+            tb_previous_decriminalization_2,
+            tb_previous_criminalization_2,
+        ],
+        ignore_index=True,
+        short_name=paths.short_name,
+    )
+
+    # Fill tb with data from START_YEAR for each country
+    tb = fill_countries_to_start_year(tb=tb)
+
+    # Fill empty status of the "Russian Empire" as "Legal" (See P8 of paper)
+    tb.loc[
+        (
+            tb["country"].isin(
+                [
+                    "Russia",
+                    "Estonia",
+                    "Latvia",
+                    "Lithuania",
+                    "Moldova",
+                    "Armenia",
+                    "Azerbaijan",
+                    "Georgia",
+                    "Kazakhstan",
+                    "Kyrgyzstan",
+                    "Tajikistan",
+                    "Turkmenistan",
+                    "Uzbekistan",
+                ]
+            )
+        )
+        & (tb["status"].isna()),
+        "status",
+    ] = "Legal"
+
+    # Keep only necesary columns
+    tb = tb[["country", "year", "status"]]
+
+    return tb
+
+
+def expand_table(tb: Table, start_col: str, end_col: str, status: str) -> Table:
+    """
+    Expand the table to have a row for each year between last_decriminalization and last_criminalization.
+    """
+
+    tb_ = tb.copy()
+    tb_ = tb_[["country", start_col, end_col]]
+
+    # Remove rows with missing values in start_col or end_col
+    tb_ = tb_.dropna(subset=[start_col, end_col])
+
+    # Create a long table that has a year column defined between last_decriminalization and last_criminalization for each country
+    tb_expanded = Table()
+
+    for country in tb_["country"].unique():
+        country_data = tb_[tb_["country"] == country].copy()
+
+        # Create a table with the years between last_decriminalization and last_criminalization
+        if end_col == "end_year":
+            years = list(range(country_data[start_col].iloc[0], country_data[end_col].iloc[0] + 1))
+        else:
+            years = list(range(country_data[start_col].iloc[0], country_data[end_col].iloc[0]))
+        country_data = country_data.reindex(country_data.index.repeat(len(years)))
+        country_data["year"] = years
+
+        tb_expanded = pr.concat([tb_expanded, country_data], ignore_index=True)
+
```

Comment:
I'd suggest working with a list, like `tbs = []` and append tables to the list instead

```py
tbs = []
for country in tb_["country"].unique():
    ...
    tbs.append(country_data)

tb_expanded = pr.concat(tbs, ignore_index=True)